src/librustc_mir/borrow_check/type_check/mod.rs

```diff
@@ -503,6 +490,31 @@ impl<'a, 'b, 'tcx> TypeVerifier<'a, 'b, 'tcx> {
             place_ty = self.sanitize_projection(place_ty, elem, place, location)
         }
 
+        if let PlaceContext::NonMutatingUse(NonMutatingUseContext::Copy) = context {
```

Comment:
The main change here is that we now do this for *all* places, right? I wonder why we didn't before? That just seems like a silly oversight to me.

Reply:
yes